src/components/tab-side-menu/tab-side-menu.tsx

```diff
@@ -60,27 +63,24 @@ const topHeaderStyle = cva("flex w-full items-center justify-between", {
 export default function TabSideMenu() {
   const [isTabMinimized, setIsTabMinimized] = useState(false);
   const [isAdding, setIsAdding] = useState(false);
-  const [goals, setGoals] = useState<TabItem[]>([]);
   const [newGoal, setNewGoal] = useState("");
+  const { mutate } = useCreateGoal();
+  const { data } = useSuspenseInfiniteQuery(getInfiniteGoalsOptions({}));
 
   const handleSubmit = (e: React.FormEvent) => {
     e.preventDefault();
     if (newGoal.trim() === "") return;
-    const newTab: TabItem = {
-      id: Date.now(),
-      text: newGoal,
+    const newGoalData: GoalResponse = {
+      id: 0,
+      title: newGoal,
+      createdAt: new Date().toISOString(),
+      updatedAt: new Date().toISOString(),
     };
-    setGoals([...goals, newTab]);
-    setNewGoal("");
-  };
-
-  // 목표 텍스트 변경 함수
-  const handleInputChange = (id: number, newValue: string) => {
-    setGoals((prevGoals) =>
-      prevGoals.map((goal) =>
-        goal.id === id ? { ...goal, text: newValue } : goal,
-      ),
-    );
+    mutate(newGoalData, {
+      onSuccess: () => {
+        setNewGoal("");
+      },
+    });
```

Comment:
여기서 아래처럼 title만 넘겨주도록 수정하는 것이 더 좋을 것 같습니다.

```suggestion
    mutate({ title: newGoal }, {
      onSuccess: () => {
        setNewGoal("");
      },
    });
```